src/storage-client/src/source/persist_source.rs

```diff
@@ -143,15 +155,66 @@ where
         Arc::new(metadata.relation_desc),
         Arc::new(UnitSchema),
         move |stats| {
-            mfp_pushdown.as_ref().map_or(true, |x| {
-                x.should_fetch(&PersistSourceDataStatsImpl { desc: &desc, stats })
-            })
+            if let Some(plan) = &filter_plan {
+                let stats = PersistSourceDataStatsImpl {
+                    desc: &fake_desc,
+                    stats,
+                };
+                filter_may_match(desc.typ().clone(), time_range.clone(), stats, plan)
+            } else {
+                true
+            }
         },
     );
     let rows = decode_and_mfp(&fetched, &name, until, map_filter_project, yield_fn);
     (rows, token)
 }
 
+fn filter_may_match(
```

Comment:
I mentioned in my last round of comments that we might want to later move this method into `mz_expr` at some point... but I just realized that maybe you are ahead of me here! :D

I intended for the `PersistSourceDataStats` trait to be the API boundary between persist and compute, but looking at this again, I had a 💡 go on and realized that the interpreter stuff is (as of this PR) effectively the API boundary instead. which might be a nicer API boundary! is that what you've been thinking?